### DISCUSSION TITLE
remeshing_isotropic_explicit_remeshing filter - Doubts on input parameters

### DISCUSSION THREAD

**Discussion by ghost** [cnr-isti-vclab/PyMeshLab#113] [Q&A] [answered]
Let's consider the file bunny10k.ply available [here](https://github.com/cnr-isti-vclab/meshlab/tree/master/sample) which has 6614 vertices and 9999 faces.

I am using MeshLab v2020.12 and pymeshlab 0.2.1.

I have two questions.

**First Question**

If I open MeshLab and apply the filter "Remeshing: Isotropic Explicit Remeshing" without changing any parameter, I get a mesh having 10827 vertices and 21522 faces as you can see in the image below.

![image](https://user-images.githubusercontent.com/85804972/121786528-a22dc500-cbc0-11eb-9889-d0ce78ab41fa.png)

If I perform the same operation using pymeshlab (so the default parameters are not modified), I get a different result, that is I get a mesh having 10803 vertices and 21474 faces as you can see in the bloew. I have also pasted the code below.
![image](https://user-images.githubusercontent.com/85804972/121786563-cab5bf00-cbc0-11eb-8fc7-ebaff14d0c6d.png)

```
import pymeshlab

ms = pymeshlab.MeshSet()
mesh_non_isotropic = ms.load_new_mesh('bunny10k.ply')
mesh_isotropic = ms.remeshing_isotropic_explicit_remeshing()
ms.save_current_mesh('bunny_isotropic_pymeshlab.ply')
```

If I take a look at the [documentation ](https://pymeshlab.readthedocs.io/en/latest/filter_list.html#remeshing_isotropic_explicit_remeshing)for this filter, the default values are equal to the MeshLab user interface default values.

Thus, I wonder what is causing this difference. Without changing any parameter both in MeshLab and pymeshlab, I expect the two meshes to be equal. For sure I am doing something wrong, but I can't spot the error.

**Second Question**

Furthermore, I am having troubles understanding how to enter the "targetlen" and "maxsurfdist" values (so when I want to change the default values). In the documentation these values are defined as percentages, but if I enter in pymeshlab the same value that I enter in the user interface (percentage field) I get two totally different meshes.

Thanks!

**Comment by alemuntoni** [comment, marked as answer]
For the first question, there could be two main reasons:
- (more likely) since python stores floating point values only with double precision, we decided to use the double precision version of meshlab as core of pymeshlab (2021.05**d**). Some algorithms may give different results if scalars are stored with double precision;
- (less likely) the widget window of filters tends to truncate floating point values that are not fully displayed in the GUI; I've already experienced this issue and I'll try to make sure that in future versions, when values are unchanged in the GUI, also the actual parameters given to the filter stays unchanged.

For the second question, if you pass as parameter a float value, it is interpreted as an absolute value. You need to pass a [Percentage](https://pymeshlab.readthedocs.io/en/latest/classes/percentage.html) object. You can find an explicative example [here](https://pymeshlab.readthedocs.io/en/latest/tutorials/apply_filter_parameters_percentage.html).

**Reply by ghost** [reply]
Dear @alemuntoni,
thanks!
Sorry, I missed the part of the documentation about the Percentage class.
Have a nice day
